api/api.go

```diff
@@ -34,12 +40,34 @@ func Start(elrondFacade MainApiHandler) error {
 	if err != nil {
 		return err
 	}
-	registerRoutes(ws, elrondFacade)
+	registerRoutes(ws, elrondFacade, elrondFacade.PrometheusMonitoring())
+
+	if elrondFacade.PrometheusMonitoring() {
+		err = joinMonitoringSystem(elrondFacade.RestApiPort(), elrondFacade.PrometheusJoinURL())
+		if err != nil {
+			return err
+		}
+	}
 
 	return ws.Run(fmt.Sprintf(":%s", elrondFacade.RestApiPort()))
 }
 
-func registerRoutes(ws *gin.Engine, elrondFacade middleware.ElrondHandler) {
+func joinMonitoringSystem(port string, prometheusJoinURL string) error {
+	req, err := http.NewRequest("POST", prometheusJoinURL, strings.NewReader(port))
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	err = resp.Body.Close()
+	if err != nil {
```

Comment:
you can directly return the err, without any extra conditional

Reply:
done